File: tree_01.c
// 一维数组存储二叉树
#include<stdio.h>

void btree_create(int *btree, int *data, int length)
{
    for (int i = 0; i < length; i++) {
        int level = 0;
        while (btree[level] != 0) {
            if (data[i] > btree[level])
                level = 2 * level + 2;
            else 
                level = 2 * level + 1;
        }
        btree[level] = data[i];
    }
}

void print_array(int *ptr, int length)
{
    for (int i = 0; i < length; i++)
        printf("[%d]  ", ptr[i]);
    printf("\n");
}

int main(void)
{
    int length = 9;
    int data[9] = {6, 3, 5, 4, 7, 8, 9, 2, 1};
    int btree[15] = {0};
    btree_create(btree, data, length);
    print_array(data, 9);
    print_array(btree, 15);
    return 0;
}

File: tree_01.py
# 一维数组存储二叉树
def Btree_create(btree, data, length):
    for i in range(1, length):
        level = 1
        while btree[level] != 0:
            if data[i] > btree[level]:
                level = level * 2 + 1
            else:
                level = level * 2
        btree[level] = data[i]

length = 9
data = [0, 6, 3, 5, 4, 7, 8, 9, 2]
btree = [0] * 16
print('original array contents:')
for i in range(length):
    print('[%2d]' %data[i], end ='  ')
print()
Btree_create(btree, data, 9)
print('Binary tree contents:')
for i in range(1, 16):
    print('[%2d]' % btree[i], end = '  ')
print()